src/Stratis.Bitcoin/Consensus/Validators/BlockValidator.cs

```diff
@@ -37,39 +48,80 @@ public interface IHeaderValidator
         void VerifyBlockIntegrity(Block block, ChainedHeader chainedHeader);
     }
 
-    public interface IPartialValidation
-    {
-        /// <summary>
-        /// Partial validation of a block, this will not changes any state in the consensus store when validating a block.
-        /// </summary>
-        /// <param name="chainedHeaderBlock">The block to validate.</param>
-        /// <param name="onPartialValidationCompletedAsyncCallback">A callback that is called when validation is complete.</param>
-        void StartPartialValidation(ChainedHeaderBlock chainedHeaderBlock, OnPartialValidationCompletedAsyncCallback onPartialValidationCompletedAsyncCallback);
-    }
-
     // <inheritdoc />
     public class HeaderValidator : IHeaderValidator
     {
         // <inheritdoc />
         public void ValidateHeader(ChainedHeader chainedHeader)
         {
-            throw new System.NotImplementedException();
+            // TODO: add rule attributes for header validation
         }
+    }
 
+    // <inheritdoc />
+    public class IntegrityValidator : IIntegrityValidator
+    {
         // <inheritdoc />
         public void VerifyBlockIntegrity(Block block, ChainedHeader chainedHeader)
         {
-            throw new System.NotImplementedException();
+            // TODO: add rule attributes for partial validation
         }
     }
 
     // <inheritdoc />
     public class PartialValidation : IPartialValidation
     {
+        private readonly IConsensusRules consensusRules;
+        private readonly AsyncQueue<PartialValidationItem> asyncQueue;
+        private readonly ILogger logger;
+
+        public PartialValidation(IConsensusRules consensusRules, ILoggerFactory loggerFactory)
+        {
+            this.consensusRules = consensusRules;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+
+            this.asyncQueue = new AsyncQueue<PartialValidationItem>(this.OnEnqueueAsync);
+        }
+
+        private async Task OnEnqueueAsync(PartialValidationItem item, CancellationToken cancellationtoken)
+        {
+            this.logger.LogTrace("({0}:'{1}')", nameof(item), item);
+
+            var validationContext = new ValidationContext {Block = item.ChainedHeaderBlock.Block};
+
+            await this.consensusRules.PartialValidationAsync(new ValidationContext { Block = item.ChainedHeaderBlock.Block }, null);
+
+            var partialValidationResult = new PartialValidationResult
+            {
+                ChainedHeaderBlock = item.ChainedHeaderBlock,
+                BanDurationSeconds = validationContext.BanDurationSeconds,
+                Succeeded = validationContext.Error != null
+            };
+
+            await item.PartialValidationCompletedAsyncCallback(partialValidationResult);
+
+            this.logger.LogTrace("(-):{0}", partialValidationResult);
```

Comment:
There is no return value from this method, so there should be no value in the exit log.